appleauth/fetch_test.go

```diff
@@ -48,58 +85,100 @@ func TestSelect(t *testing.T) {
 			args: args{
 				devportalConnection: &devportalservice.AppleDeveloperConnection{},
 				authSources:         []Source{&ConnectionAPIKeySource{}, &ConnectionAppleIDSource{}, &InputAPIKeySource{}, &InputAppleIDSource{}},
-				inputs: Inputs{
-					Username: "a", Password: "b", AppSpecificPassword: "c",
-					APIIssuer: "", APIKeyPath: "",
+				inputs:              argInput,
+			},
+			want: Credentials{
+				AppleID: &expectedAppleIDWithArgInput,
+				APIKey:  nil,
+			},
+		},
+		{
+			name: "Connection active (Apple ID), inputs (Apple ID) with ConnectionAppleIDSource",
+			args: args{
+				devportalConnection: &devportalservice.AppleDeveloperConnection{
+					AppleIDConnection: &argAppleIDConnection,
 				},
+				authSources: []Source{&ConnectionAppleIDSource{}},
+				inputs:      argInput,
+			},
+			want: Credentials{
+				AppleID: &expectedAppleIDWithArgConnection,
+				APIKey:  nil,
+			},
+		},
+		{
+			name: "Connection active (Apple ID), inputs (Apple ID) with InputAppleIDSource",
+			args: args{
+				devportalConnection: &devportalservice.AppleDeveloperConnection{
+					AppleIDConnection: &argAppleIDConnection,
+				},
+				authSources: []Source{&InputAppleIDSource{}},
+				inputs:      argInput,
+			},
+			want: Credentials{
+				AppleID: &expectedAppleIDWithArgInput,
+				APIKey:  nil,
+			},
+		},
+		{
+			name: "Connection active (Apple ID), inputs (Apple ID) with ConnectionAppleIDFastlaneSource",
+			args: args{
+				devportalConnection: &devportalservice.AppleDeveloperConnection{
+					AppleIDConnection: &argAppleIDConnection,
+				},
+				authSources: []Source{&ConnectionAppleIDFastlaneSource{}},
+				inputs:      argInput,
 			},
 			want: Credentials{
 				AppleID: &AppleID{
-					Username: "a", Password: "b", AppSpecificPassword: "c", Session: "",
+					Username:            argAppleIDConnection.AppleID,
+					Password:            argAppleIDConnection.Password,
+					AppSpecificPassword: argInput.AppSpecificPassword,
+					Session:             "",
 				},
 				APIKey: nil,
 			},
 		},
+		{
```

Comment:
Can you please add a test where the Connection is used, but the app specific password is used from the input?

Reply:
Giving it a second thought, how is this scenario different from the `Connection active (Apple ID), inputs (Apple ID) with ...` ones?

Reply:
There is an app specific password in the argAppleIDConnection struct, I do not get why it is not used in this case

Reply:
I'll DM you